### PR TITLE
doc: update using grub tutorial

### PR DESCRIPTION
**Pull request by ZideChen0**
Signed-off-by: Zide Chen <zide.chen@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/using_grub.rst [hunk 1]
```diff
@@ -191,3 +192,35 @@ Here we provide another simple method to build GRUB in efi application format:
...
+This is because by default GRUB loads multiboot2 modules to host physical address
+starting from 1MB, which may be overlapped with the SOS Linux kernel preferred
+address which is confiugred at 16MB by default. (``CONFIG_PHYSICAL_START`` in
+Linux kernel config).
+
+This can be resolved by enabling ``quirk-modules-after-kernel`` in GURB and
+change the multiboot2 load command. Refer to :ref:`self-built-grub` section to
+download, build and install GRUB boot loader with the following extra steps:
+
+# Apply the following patch before building GRUB.
```
Comment (dbkinder) [inline doc/tutorials/using_grub.rst:211]: Change to:
```
#. Download and apply the following patch before building GRUB:
```

File: doc/tutorials/using_grub.rst [hunk 2]
```diff
@@ -191,3 +192,35 @@ Here we provide another simple method to build GRUB in efi application format:
...
+
+   `multiboot2-Implement-quirk-modules-after-kernel.patch
+   <../_static/download/multiboot2-Implement-quirk-modules-after-kernel.patch>`_
```
Comment (dbkinder) [inline doc/tutorials/using_grub.rst:214]: should be ``downloads`` (plural)

File: doc/tutorials/using_grub.rst [hunk 3]
```diff
@@ -191,3 +192,35 @@ Here we provide another simple method to build GRUB in efi application format:
 #. Copy ACRN binary and guest kernel images to the GRUB-configured folder, e.g. ``/boot/`` folder on ``/dev/sda3/``;
 
 #. Run ``/EFI/boot/grub_x86_64.efi`` in the EFI shell.
+
+Enable quirk-modules-after-kernel for Multiboot2
+************************************************
+
+With the released GRUB (up to v2.06), SOS may not boot if the size of the SOS
```
Comment (dbkinder) [inline doc/tutorials/using_grub.rst:199]: change ``SOS`` to ``the Service VM may not boot if the size of the Service VM OS``

**Comment by NanlinXie** [comment]
Set it as on hold since we are trying another approach to solve this issue without forcing user to reinstall the customized grub.

File: doc/tutorials/using_grub.rst [hunk 4]
```diff
@@ -191,3 +192,35 @@ Here we provide another simple method to build GRUB in efi application format:
...
+This can be resolved by enabling ``quirk-modules-after-kernel`` in GURB and
+change the multiboot2 load command. Refer to :ref:`self-built-grub` section to
+download, build and install GRUB boot loader with the following extra steps:
+
+# Apply the following patch before building GRUB.
+
+   `multiboot2-Implement-quirk-modules-after-kernel.patch
+   <../_static/download/multiboot2-Implement-quirk-modules-after-kernel.patch>`_
+
+# In ``grub.cfg``, append ``--quirk-modules-after-kernel`` to the multiboot2
```
[the remark this answers is not in the document]
Comment (ZideChen0) [reply doc/tutorials/using_grub.rst:216]: Thanks, will do.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by dbkinder** [comment]
@ZideChen0 I'm told by @NanlinXie that this PR can be closed without merging (a different solution is in play)

**Pull request closed without merging**